examples/generative/speaker_recognition_using_cnn.py

```diff
@@ -0,0 +1,520 @@
+
+# Supported audio file format
+FILE_FORMATS = ".wav"
+
+# Percentage of samples to use for validation
+VALID_SPLIT = 0.1
+
+# Seed to use when shuffling the dataset and the noise
+SHUFFLE_SEED = 43
+
+# The sampling rate to use.
+# This is the one used in all of the audio samples.
+# We will resample all of the noise to this sampling rate.
+# This will also be the output size of the audio wave samples
+# (since all samples are of 1 second long)
+DESIRED_SR = 16000
+
+# The factor to multiply the noise with according to:
+#   noisy_sample = sample + noise * prop * scale
+#      where prop = sample_amplitude / noise_amplitude
+SCALE = 0.5
+
+BATCH_SIZE = 128
+EPOCHS = 1
+
+"""
+## Data Preparation
+"""
+
+"""
+The dataset is composed of 7 folders that can be divided into 2 groups:
+- One for the speeches containing 5 folders for 5 different speakers. Each folder contain
+1500 audio files each of which is 1 second long and sampled at 16000 samples per second
+- One for different noises containing 2 folders and an overall of 6 files. Those files
+are longer than 1 second (and originally not sampled at 16000, but we assume that by at
+this stage, you would have resampled them to 16000). We will use those 6 files to create
+354 1-second-long noise samples to be used for training
+"""
+
+# The dataset is composed of 7 folders that can be divided into 2 groups:
+# - Speakers' audio folders:
+#   5 folders containing each 1500 audio files each of which is 1 second long
+#   and sampled at 16000 samples per second
+# - Noise folders:
+#   2 folders and an overall of 6 noise audio files.
+#   These files are longer than 1 second and not all of them sampled at 16000.
+#   We will use those 6 files to create 354 1-second-long noise samples
+
+# Let's separate these 2 categories into 2 folders:
+# - 'audio' which will contain all the speakers' audio folders and files
+# - 'noise' which will contain all the noise sounds and folders
+
+if os.path.exists(DATASET_AUDIO_PATH) is False:
+    os.makedirs(DATASET_AUDIO_PATH)
+
+if os.path.exists(DATASET_NOISE_PATH) is False:
+    os.makedirs(DATASET_NOISE_PATH)
+
+if (
+    len(os.listdir(DATASET_AUDIO_PATH)) == 0
+    and len(os.listdir(DATASET_NOISE_PATH)) == 0
+):
+    for folder in os.listdir(DATASET_ROOT):
+        if os.path.isdir(os.path.join(DATASET_ROOT, folder)):
+            if folder in [AUDIO_SUBFOLDER, NOISE_SUBFOLDER]:
+                continue
+            elif folder in NOISE_FOLDERS:
+                shutil.move(
+                    os.path.join(DATASET_ROOT, folder),
+                    os.path.join(DATASET_NOISE_PATH, folder),
+                )
+            else:
+                shutil.move(
+                    os.path.join(DATASET_ROOT, folder),
+                    os.path.join(DATASET_AUDIO_PATH, folder),
+                )
+
+"""
+## Noise Preparation
+"""
+
+# Let's first get the list of all noise files
+noise_paths, _, _ = tf.python.keras.preprocessing.dataset_utils.index_directory(
```

Comment:
This is a private API, do not use it

Reply:
Sorry, I did not know I shouldn't use it. I implemented the bits that I want.

Reply:
In general, do not import anything from `tensorflow.python`, stick to the public API